cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/FullModel.java

```diff
@@ -0,0 +1,208 @@
+        return modelingAuthoritySet;
+    }
+
+    @Override
+    public String toString() {
+        return "FullModel(" +
+                "id='" + id + '\'' +
+                ", scenarioTime=" + scenarioTime +
+                ", created=" + created +
+                ", description='" + description + '\'' +
+                ", version=" + version +
+                ", profiles=" + profiles +
+                ", dependentOn=" + dependentOn +
+                ", supersedes=" + supersedes +
+                ", modelingAuthoritySet='" + modelingAuthoritySet + '\'' +
+                ')';
+    }
+
+    private static class ParsingContext {
+
+        private String id;
+
+        private ZonedDateTime scenarioTime;
+
+        private ZonedDateTime created;
+
+        private String description;
+
+        private Integer version;
+
+        private final List<String> profiles = new ArrayList<>();
+
+        private final List<String> dependentOn = new ArrayList<>();
+
+        private final List<String> supersedes = new ArrayList<>();
+
+        private String modelingAuthoritySet;
+
+        private void validate() {
+            if (id == null) {
+                throw new PowsyblException("ID is missing");
+            }
+            if (scenarioTime == null) {
+                throw new PowsyblException("Scenario time is missing");
+            }
+            if (created == null) {
+                throw new PowsyblException("Created time is missing");
+            }
+            if (version == null) {
+                throw new PowsyblException("Version is missing");
+            }
+            if (profiles.isEmpty()) {
+                throw new PowsyblException("At least one profile is required");
+            }
+            if (modelingAuthoritySet == null) {
+                throw new PowsyblException("Modeling authority set is missing");
+            }
+        }
+    }
+
+    public static FullModel parse(Reader reader) {
+        Objects.requireNonNull(reader);
+        ParsingContext context = new ParsingContext();
+        try {
+            XMLInputFactory factory = XMLInputFactory.newInstance();
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+
+            XMLStreamReader xmlReader = factory.createXMLStreamReader(reader);
+            try {
+                XmlUtil.readUntilStartElement(new String[] {"/", CgmesNames.RDF, CgmesNames.FULL_MODEL}, xmlReader, () -> {
+                    context.id = xmlReader.getAttributeValue(CgmesNamespace.RDF_NAMESPACE, CgmesNames.ABOUT);
+                    XmlUtil.readUntilEndElement(CgmesNames.FULL_MODEL, xmlReader, () -> {
+                        switch (xmlReader.getLocalName()) {
+                            case CgmesNames.SCENARIO_TIME:
+                                context.scenarioTime = ZonedDateTime.parse(XmlUtil.readUntilEndElement(CgmesNames.SCENARIO_TIME, xmlReader, () -> { }));
```

Comment:
You should introduce an overload for readUntilElement, that takes only the two first parameters. This would make this code more readable:
```
XmlUtil.readUntilEndElement(CgmesNames.SCENARIO_TIME, xmlReader);
```

This overload would call the existing one, with the empty lambda

Another idea would be to name this method `XmlUtil.readText()`.

Reply:
Ok for XmlUtil.readText()

Reply:
Done and sync with the other PR